flopy/utils/get_modflow.py

```diff
@@ -291,58 +363,12 @@ def run_main(
         raise KeyError(
             f"repo {repo!r} not supported; choose one of {available_repos}"
         )
-    api_url = f"https://api.github.com/repos/{owner}/{repo}"
-
-    if release_id == "latest":
-        req_url = f"{api_url}/releases/latest"
-    else:
-        req_url = f"{api_url}/releases/tags/{release_id}"
-    request = get_request(req_url)
-    avail_releases = None
-    num_tries = 0
-    while True:
-        num_tries += 1
-        try:
-            with urllib.request.urlopen(request, timeout=10) as resp:
-                result = resp.read()
-                remaining = int(resp.headers["x-ratelimit-remaining"])
-                if remaining <= 10:
-                    print(
-                        f"Only {remaining} GitHub API requests remaining "
-                        "before rate-limiting"
-                    )
-                break
-        except urllib.error.HTTPError as err:
-            if err.code == 401 and os.environ.get("GITHUB_TOKEN"):
-                raise ValueError("GITHUB_TOKEN env is invalid") from err
-            elif err.code == 403 and "rate limit exceeded" in err.reason:
-                raise ValueError(
-                    f"use GITHUB_TOKEN env to bypass rate limit ({err})"
-                ) from err
-            elif err.code == 404:
-                if avail_releases is None:
-                    avail_releases = get_avail_releases(api_url)
-                if release_id in avail_releases:
-                    if num_tries < max_http_tries:
-                        # GitHub sometimes returns 404 for valid URLs, so retry
-                        print(f"URL request {num_tries} did not work ({err})")
-                        continue
-                else:
-                    raise ValueError(
-                        f"Release {release_id!r} not found -- "
-                        f"choose from {avail_releases}"
-                    ) from err
-            elif err.code == 503 and num_tries < max_http_tries:
-                # GitHub sometimes returns this error for valid URLs, so retry
-                print(f"URL request {num_tries} did not work ({err})")
-                continue
-            raise RuntimeError(f"cannot retrieve data from {req_url}") from err
-
-    release = json.loads(result.decode())
-    tag_name = release["tag_name"]
-    if not quiet:
-        print(f"fetched release {tag_name!r} from {owner}/{repo}")
-
+    avail_releases = get_releases(repo, quiet)
+    if release_id not in avail_releases:
```

Comment:
this check is not necessary and prevents getting a release that is not listed by `get_releases` ([by default `per_page` is 30](https://docs.github.com/en/rest/releases/releases#list-releases), so this is not a complete list). E.g. `get-modflow --repo modflow6-nightly-build --release-id 20220914 /tmp/bin` would fail, but is a valid release. An invalid release returns a 404 error code, as previously implemented.

Reply:
Got it- should have anticipated results would be paged. Will revert and add a test case. Thanks!

Reply:
@mwtoews after some testing, it looks like the releases API only exposes the last 30 releases, regardless of pagination:

```shell
$ curl \
-H "Accept: application/vnd.github+json" \
-H "Authorization: Bearer <token>" \
"https://api.github.com/repos/MODFLOW-USGS/modflow6-nightly-build/releases?per_page=100" | jq '.[].tag_name'
...
"20221018"
"20221017"
"20221016"
"20221015"
"20221014"
"20221013"
"20221012"
"20221011"
"20221010"
"20221009"
"20221008"
"20221007"
"20221006"
"20221005"
"20221004"
"20221003"
"20221002"
"20221001"
"20220930"
"20220929"
"20220928"
"20220927"
"20220926"
"20220925"
"20220924"
"20220923"
"20220922"
"20220921"
"20220920"
"20220919"
"20220918"
```

Trying to retrieve a valid release from further back fails:

```shell
$ curl \
-H "Accept: application/vnd.github+json" \
-H "Authorization: Bearer <token>" \
https://api.github.com/repos/MODFLOW-USGS/modflow6-nightly-build/releases/tags/20220917
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/repos#get-a-release-by-tag-name"
}
```

I haven't found this 30-release policy documented anywhere &mdash; if I'm unable to after a bit more searching I might consider opening a PR to mention it [on this page](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases).

I've still updated the code here to only check available releases if a release with the given tag is not found, cutting out an unnecessary request. Let me know if I'm missing something or anything else should be changed. Thanks again for the reviews

Reply:
@mwtoews please ignore above- realized it's not a GitHub policy, the nightly build CI deletes releases older than 30 days. Sorry about that.